layer/store/geo.ts

```diff
@@ -0,0 +1,207 @@
+          `api/${sharedGeoStore.ipAddress}`,
+          {
+            key: PROXY_DETECTION_API_KEY
+          }
+        )) as {
+          data: {
+            security: {
+              vpn: boolean
+              proxy: boolean
+              tor: boolean
+              relay: boolean
+            }
+            location: {
+              country_code: string
+            }
+          }
+        }
+
+        if (!response.data) {
+          sharedGeoStore.$patch({
+            vpnDetected: false
+          })
+
+          return
+        }
+
+        const { security } = response.data
+
+        const vpnDetected =
+          security.proxy || security.vpn || security.tor || security.relay
+
+        sharedGeoStore.$patch({
+          vpnDetected
+        })
+      } catch (e: unknown) {
+        sharedGeoStore.$patch({
+          vpnDetected: false
+        })
+      }
+    },
+
+    async fetchUserCountryFromBrowser() {
+      const sharedGeoStore = useSharedGeoStore()
+
+      const position = (await new Promise((resolve, reject) =>
+        navigator.geolocation.getCurrentPosition(resolve, reject)
+      ).catch(() => {
+        return {
+          longitude: '',
+          latitude: ''
+        }
+      })) as {
+        coords: {
+          longitude: string
+          latitude: string
+        }
+      }
+
+      if (position.coords.longitude === '' || position.coords.latitude === '') {
+        return
+      }
+
+      const googleMapsHttpClient = new HttpClient(
+        'https://maps.googleapis.com/maps/api/geocode/'
+      )
+      const GOOGLE_MAPS_SUFFIX = `json?latlng=${position.coords.longitude},${position.coords.latitude}&sensor=false&key=${GOOGLE_MAPS_KEY}`
+
+      try {
+        const response = (await googleMapsHttpClient.get(
+          GOOGLE_MAPS_SUFFIX
+        )) as {
+          data: {
+            results: {
+              address_components: { types: string[]; short_name: string }[]
+            }[]
+          }
+        }
+
+        const [results] = response.data.results
+
+        const country = results.address_components.find((component) =>
+          component.types.includes('country')
+        )
+
+        sharedGeoStore.$patch({
+          browserCountry: country?.short_name || ''
+        })
+      } catch (e: unknown) {
+        // silently throw
+      }
+    },
+
+    async fetchUserLocation() {
+      const sharedGeoStore = useSharedGeoStore()
+
+      await sharedGeoStore.fetchGeoLocation()
+
+      if (VPN_CHECKS_ENABLED) {
+        const todayInSeconds = Math.floor(Date.now() / 1000)
+
+        await sharedGeoStore.fetchVPNStatus()
+
+        if (!sharedGeoStore.vpnDetected) {
+          return
+        }
+
+        const shouldCheckVpnOrProxyUsage = SECONDS_IN_A_DAY.times(7)
+          .plus(sharedGeoStore.vpnCheckedTimestamp)
+          .lte(todayInSeconds)
+
+        if (!shouldCheckVpnOrProxyUsage) {
+          return
+        }
+
+        await sharedGeoStore.fetchUserCountryFromBrowser()
```

Comment:
user needs to allow location access in the browser and system settings in order to fetch the user's location from the browser

so we should resurrect the following function and have it run before line `199` imo
```
export const displayVPNOrProxyUsageToast = () => {
  const notificationStore = useSharedNotificationStore()

  const MORE_INFO_URL =
    'https://helixapp.zendesk.com/hc/en-us/articles/5377904870415-Who-can-use-Helix-'
  const TOAST_DURATION = 10 * 1000

  notificationStore.info({
    title: 'VPN or proxy detected',
    description:
      'Please make sure that you have allowed location access in your browser and system settings.',
    timeout: TOAST_DURATION,
    actions: [
      {
        key: MORE_INFO_URL,
        label: 'Learn More',
        callback: () => window.open(MORE_INFO_URL, '_blank')
      }
    ]
  })
}
```